hax-lib/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+//! Hax-specific helpers for Rust programs. Those helpers are usually
+//! no-ops when compiled normally but meaningful when compiled under
+//! hax.
```

Comment:
Let's add examples (and thus tests) for everything.
Either here or at the macros.

```suggestion
//! hax.
//!
//! # Example:
//!
//! ```rust
//! fn sum(x: u32, y: u32) -> u32 {
//!   hax_lib::assert!(x < 4242 && y < 424242);
//!   hax_lib::debug_assert!(x < 4242 && y < 424242);
//!   x + y
//! }
//! ```
```